internal/services/v1/watch_test.go

```diff
@@ -162,16 +159,18 @@ func TestWatch(t *testing.T) {
 				require.NoError(err)
 
 				var receivedUpdates []*v1.RelationshipUpdate
-				select {
-				case updates := <-updatesChan:
-					receivedUpdates = updates
-				case <-time.After(3 * time.Second):
-					require.FailNow("timed out waiting for updates")
-					return
-				}
-
-				require.Equal(len(tc.expectedUpdates), len(receivedUpdates))
 
+				for len(receivedUpdates) < len(tc.expectedUpdates) {
+					select {
+					case updates := <-updatesChan:
+						receivedUpdates = append(receivedUpdates, updates...)
+						fmt.Printf("expected: %v\n", tc.expectedUpdates)
+						fmt.Printf("received: %v\n", receivedUpdates)
+					case <-time.After(1 * time.Second):
+						require.FailNow("timed out waiting for updates")
+						return
+					}
+				}
```

Comment:
We should check the updates received match, content wise, those expected

Reply:
There is no canonical ordering for the v1.RelationshipUpdates, so testing this becomes a major effort.

Reply:
Can we sort and see that we've at least seen the expected ones?

Reply:
Yes of course it's possible, but it raises the question: why was the test accepted in the first place only checking len?

Reply:
Done.

Reply:
Probably an oversight in my review